### DISCUSSION TITLE
What does nil mean, in the context of true, false and nil?

### DISCUSSION THREAD

**Discussion by JoeDaBu** [rmagatti/auto-session#288] [Q&A] [answered]
For auto_save_enabled and some other settings they have options of false, true, nil, with the default being set to nil. The description says Enables/disables auto saving, ok so how are there three values for enabling or disabling auto saving. Does nil then mean false? What is the importance of including nil, in a true or false statement?

I understand what nil does for other settings that take in lists, but I don't understand how its applied here. Especially given some other settings only have true or false options.

**Comment by rmagatti** [comment, marked as answer]
`nil` would be the same as not passing the option at all, which means the config would just use default options. I suspect the handling for that might not be correct for all of these configs yet but that's the intention.
